app/scripts/services/validateRegistrant.js

```diff
@@ -194,49 +209,49 @@ angular
       return false;
     };
 
-    this.blockVisible = function(block, registrant, isAdmin) {
-      let visible =
+    this.blockVisible = (block, registrant, isAdmin) => {
+      const visible =
         angular.isDefined(registrant) &&
         blockVisibleRuleCheck(
           block,
           registrant,
           ruleTypeConstants.SHOW_QUESTION,
         ) &&
-        this.blockInRegistrantType(block, registrant) &&
+        blockInRegistrantType(block, registrant) &&
         this.isAnyChoiceVisible(block, registrant);
 
       return block.adminOnly && !isAdmin ? false : visible;
     };
 
-    this.choiceVisible = function(block, choice, registrant) {
+    this.choiceVisible = (block, choice, registrant) => {
       return (
         angular.isDefined(registrant) &&
         choiceVisibleRuleCheck(
           block,
           choice,
           registrant,
           ruleTypeConstants.SHOW_OPTION,
-        ) &&
-        this.blockInRegistrantType(block, registrant)
+        )
       );
     };
 
-    this.checkboxDisable = function(block, registrant) {
+    this.checkboxDisable = (block, registrant) => {
       return blockVisibleRuleCheck(
         block,
         registrant,
         ruleTypeConstants.FORCE_SELECTION,
       );
     };
 
-    this.validate = function(conference, registrant, page) {
-      var invalidBlocks = [];
+    this.validate = (conference, registrant, page) => {
+      let invalidBlocks = [];
+
       conference = angular.copy(conference);
-      var blocks = page
+      const blocks = page
         ? _.find(conference.registrationPages, { id: page }).blocks
         : _.flatten(_.map(conference.registrationPages, 'blocks'));
-      var that = this;
-      angular.forEach(blocks, function(block) {
+
+      _.map(blocks, block => {
```

Comment:
I love `map` but if you aren't using the return value, `_.forEach` is probably better.

Reply:
Yeah, you're right. I will convert this back